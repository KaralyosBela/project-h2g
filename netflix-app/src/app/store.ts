import { configureStore } from "@reduxjs/toolkit";
import moviesReducer from "../features/moviesSlice";
import hehe from "../features/actions"

export const store = configureStore({
  reducer: {
    movies: moviesReducer,
    bruh: hehe
  }
});

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
