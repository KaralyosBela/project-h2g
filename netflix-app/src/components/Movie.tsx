import { useState } from "react";
import { IMovies } from "../interfaces/movies.interface";
import { DeleteMovieModal } from "./DeleteMovieModal";
import { EditMovieModal } from "./EditMovieModal";
import classes from "./Movie.module.css";
import {deleteMovie} from "../features/moviesSlice";
import { AppDispatch } from "../app/store";
import { useDispatch } from "react-redux";

interface Props {
  movie: IMovies;
}

export const Movie: React.FC<Props> = ({ movie }) => {
  const [modalOpen, setModalOpen] = useState<boolean>(false);
  const [editModalOpen, setEditModalOpen] = useState<boolean>(false);
  const [deleteModalOpen, setDeleteModalOpen] = useState<boolean>(false);

  const dispatch = useDispatch<AppDispatch>();

  const toggleModal = () => {
    setModalOpen(!modalOpen);
  }

  const hideModal = () => {
    setEditModalOpen(false);
    setDeleteModalOpen(false);
  }

  const delMovie = () => {
    dispatch(deleteMovie(movie))
    hideModal();
  }

  return (
    <div>
      <DeleteMovieModal hide={hideModal} show={deleteModalOpen} delMovie={delMovie}/>
      <EditMovieModal hide={hideModal} show={editModalOpen}/>
      <div className={classes.card}>
        <div className={classes.circle} onClick={toggleModal}>
          <div className={classes.firstDot}></div>
          <div className={classes.secondDot}></div>
          <div className={classes.thirdDot}></div>
        </div>
        {modalOpen && (
          <div className={classes.dropdownMenu}>
            <div className={classes.editModal} onClick={() => setEditModalOpen(!editModalOpen)}>Edit</div>
            <div className={classes.deleteModal} onClick={() => setDeleteModalOpen(!deleteModalOpen)}>Delete</div>
          </div>
        )}
        <img className={classes.image} src={movie.thumbnail} alt="alt"></img>
        <div className={classes.info}>
          <div className={classes.title}>{movie.title}</div>
          <div className={classes.year}>{movie.release_date}</div>
        </div>
        <div className={classes.genre}>{movie.genre}</div>
      </div>
    </div>
  );
};
