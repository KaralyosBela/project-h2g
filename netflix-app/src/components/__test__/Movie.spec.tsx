import { fireEvent, render, screen } from "@testing-library/react";
import { Movie } from "../Movie";
import { Provider } from "react-redux";
import { store } from "../../app/store";
import "@testing-library/jest-dom/extend-expect";

const movie = {
  id: "903",
  title: "Cool Hand Luke",
  tagline: "What we've got here is failure to communicate.",
  vote_average: 7.7,
  vote_count: 471,
  release_date: "1967-06-22",
  poster_path:
    "https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg",
  overview:
    "When petty criminal Luke Jackson is sentenced to two years in a Florida prison farm, he doesn't play by the rules of either the sadistic warden or the yard's resident heavy, Dragline, who ends up admiring the new guy's unbreakable will. Luke's bravado, even in the face of repeated stints in the prison's dreaded solitary confinement cell, \"the box,\" make him a rebel hero to his fellow convicts and a thorn in the side of the prison officers.",
  budget: 3200000,
  revenue: 16217773,
  genres: ["Crime", "Drama"],
  runtime: 126,
};

// const mockDispatch = jest.fn();
// const mockUseDispatch = jest.fn(() => mockDispatch()); //itt is megtudom hívni

// jest.mock("react-redux", () => ({
//   useDispatch: () => mockUseDispatch(),
// }));

describe("Movie component", () => {

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render Movie component with the movie details", () => {
    const { container } = render(
      <Provider store={store}>
        <Movie movie={movie} />
      </Provider>
    );
    expect(container).toMatchSnapshot();
    // screen.debug();
  });

  it("should open options modal on when clicked on the movie card", () => {
    const { container } = render(
      <Provider store={store}>
        <Movie movie={movie} />
      </Provider>
    );

    //divet hogy érdemes lekérni
    const movieCard = screen.getByTestId("card");
    fireEvent.click(movieCard);
    expect(container).toMatchSnapshot();
    // screen.debug();
  });

  it("should open options modal on when clicked the circle on the card", () => {
    const { container } = render(
      <Provider store={store}>
        <Movie movie={movie} />
      </Provider>
    );

    //divet hogy érdemes lekérni
    const circle = screen.getByTestId("card");
    fireEvent.click(circle);
    expect(container).toMatchSnapshot();
    // screen.debug();
  });

  it("should open the edit modal", () => {
    const { container } = render(
      <Provider store={store}>
        <Movie movie={movie} />
      </Provider>
    );

    //divet hogy érdemes lekérni
    const card = screen.getByTestId("card");
    fireEvent.click(card);
    const editBtn = screen.getByText(/Edit/i);
    fireEvent.click(editBtn);
    // expect(mockUseDispatch.mock.calls.length).toEqual(2);
    // expect(mockUseDispatch).toBeCalledWith(movie);
    expect(container).toMatchSnapshot();
    // screen.debug();
  });

  it("should open the delete modal", () => {
    const { container } = render(
      <Provider store={store}>
        <Movie movie={movie} />
      </Provider>
    );

    //divet hogy érdemes lekérni
    const card = screen.getByTestId("card");
    fireEvent.click(card);
    const deleteBtn = screen.getByText(/Delete/i);
    fireEvent.click(deleteBtn);
    expect(container).toMatchSnapshot();
    // screen.debug();
  });

  it("should close the options modal on card leave", () => {
    const { container } = render(
      <Provider store={store}>
        <Movie movie={movie} />
      </Provider>
    );

    //divet hogy érdemes lekérni
    const card = screen.getByTestId("card");
    fireEvent.click(card);
    fireEvent.mouseOut(card);
    expect(container).toMatchSnapshot();
    // screen.debug();
  });
});
