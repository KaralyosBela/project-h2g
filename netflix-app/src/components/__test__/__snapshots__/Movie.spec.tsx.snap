// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie component should close the options modal on card leave 1`] = `
<div>
  <div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="circle"
        data-testid="circle"
      >
        <div
          class="firstDot"
        />
        <div
          class="secondDot"
        />
        <div
          class="thirdDot"
        />
      </div>
      <img
        alt="Not found"
        class="image"
        src="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
      />
      <div
        class="info"
      >
        <div
          class="title"
        >
          Cool Hand Luke
        </div>
        <div
          class="year"
        >
          1967-06-22
        </div>
      </div>
      <div
        class="genre"
      >
        Crime, Drama
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie component should open options modal on when clicked on the movie card 1`] = `
<div>
  <div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="circle"
        data-testid="circle"
      >
        <div
          class="firstDot"
        />
        <div
          class="secondDot"
        />
        <div
          class="thirdDot"
        />
      </div>
      <div
        class="dropdownMenu"
      >
        <div
          class="infoModal"
        >
          Info
        </div>
        <div
          class="editModal"
        >
          Edit
        </div>
        <div
          class="deleteModal"
        >
          Delete
        </div>
      </div>
      <img
        alt="Not found"
        class="image"
        src="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
      />
      <div
        class="info"
      >
        <div
          class="title"
        >
          Cool Hand Luke
        </div>
        <div
          class="year"
        >
          1967-06-22
        </div>
      </div>
      <div
        class="genre"
      >
        Crime, Drama
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie component should open options modal on when clicked the circle on the card 1`] = `
<div>
  <div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="circle"
        data-testid="circle"
      >
        <div
          class="firstDot"
        />
        <div
          class="secondDot"
        />
        <div
          class="thirdDot"
        />
      </div>
      <div
        class="dropdownMenu"
      >
        <div
          class="infoModal"
        >
          Info
        </div>
        <div
          class="editModal"
        >
          Edit
        </div>
        <div
          class="deleteModal"
        >
          Delete
        </div>
      </div>
      <img
        alt="Not found"
        class="image"
        src="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
      />
      <div
        class="info"
      >
        <div
          class="title"
        >
          Cool Hand Luke
        </div>
        <div
          class="year"
        >
          1967-06-22
        </div>
      </div>
      <div
        class="genre"
      >
        Crime, Drama
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie component should open the delete modal 1`] = `
<div>
  <div>
    <div
      class="overlay"
      data-testid="overlay"
    />
    <div
      class="deleteModal"
    >
      <h1
        class="title"
      >
        DELETE MOVIE
      </h1>
      <div
        class="subtitle"
      >
        Are you sure you want to delete this movie?
      </div>
      <button>
        CONFIRM
      </button>
      <div
        class="close"
      >
        <svg
          fill="none"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.2253 4.81108C5.83477 4.42056 5.20161 4.42056 4.81108 4.81108C4.42056 5.20161 4.42056 5.83477 4.81108 6.2253L10.5858 12L4.81114 17.7747C4.42062 18.1652 4.42062 18.7984 4.81114 19.1889C5.20167 19.5794 5.83483 19.5794 6.22535 19.1889L12 13.4142L17.7747 19.1889C18.1652 19.5794 18.7984 19.5794 19.1889 19.1889C19.5794 18.7984 19.5794 18.1652 19.1889 17.7747L13.4142 12L19.189 6.2253C19.5795 5.83477 19.5795 5.20161 19.189 4.81108C18.7985 4.42056 18.1653 4.42056 17.7748 4.81108L12 10.5858L6.2253 4.81108Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="circle"
        data-testid="circle"
      >
        <div
          class="firstDot"
        />
        <div
          class="secondDot"
        />
        <div
          class="thirdDot"
        />
      </div>
      <img
        alt="Not found"
        class="image"
        src="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
      />
      <div
        class="info"
      >
        <div
          class="title"
        >
          Cool Hand Luke
        </div>
        <div
          class="year"
        >
          1967-06-22
        </div>
      </div>
      <div
        class="genre"
      >
        Crime, Drama
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie component should open the edit modal 1`] = `
<div>
  <div>
    <div
      class="overlay"
    />
    <div
      class="modal"
    >
      <h1>
        EDIT MOVIE
      </h1>
      <form>
        <div
          class="modalbody"
        >
          <div
            class="leftside"
          >
            <label
              for="title"
            >
              TITLE
            </label>
            <input
              id="title"
              type="text"
              value="Cool Hand Luke"
            />
            <label
              for="url"
            >
              MOVIE URL
            </label>
            <input
              id="url"
              type="text"
              value="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
            />
            <label
              for="genre"
            >
              GENRE
            </label>
            <div
              class=" css-b62m3t-container"
            >
              <span
                class="css-1f43avz-a11yText-A11yText"
                id="react-select-2-live-region"
              />
              <span
                aria-atomic="false"
                aria-live="polite"
                aria-relevant="additions text"
                class="css-1f43avz-a11yText-A11yText"
              />
              <div
                class=" css-18pngyw-Control"
              >
                <div
                  class=" css-319lph-ValueContainer"
                >
                  <div
                    class=" css-14el2xx-placeholder"
                    id="react-select-2-placeholder"
                  >
                    Select...
                  </div>
                  <div
                    class=" css-6j8wv5-Input"
                    data-value=""
                  >
                    <input
                      aria-autocomplete="list"
                      aria-describedby="react-select-2-placeholder"
                      aria-expanded="false"
                      aria-haspopup="true"
                      autocapitalize="none"
                      autocomplete="off"
                      autocorrect="off"
                      class=""
                      id="react-select-2-input"
                      role="combobox"
                      spellcheck="false"
                      style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                      tabindex="0"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class=" css-1hb7zxy-IndicatorsContainer"
                >
                  <span
                    class=" css-1okebmr-indicatorSeparator"
                  />
                  <div
                    aria-hidden="true"
                    class=" css-tlfecz-indicatorContainer"
                  >
                    <svg
                      aria-hidden="true"
                      class="css-tj5bde-Svg"
                      focusable="false"
                      height="20"
                      viewBox="0 0 20 20"
                      width="20"
                    >
                      <path
                        d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                      />
                    </svg>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="rightside"
          >
            <label
              for="releasedate"
            >
              RELEASE DATE
            </label>
            <input
              id="releasedate"
              type="text"
              value="1967-06-22"
            />
            <label
              for="rating"
            >
              RATING
            </label>
            <input
              id="rating"
              type="number"
              value="471"
            />
            <label
              for="runtime"
            >
              RUNTIME
            </label>
            <input
              id="runtime"
              type="number"
              value="126"
            />
          </div>
        </div>
        <div
          class="overview"
        >
          <label
            for="overview"
          >
            OVERVIEW
          </label>
          <textarea
            id="overview"
          >
            When petty criminal Luke Jackson is sentenced to two years in a Florida prison farm, he doesn't play by the rules of either the sadistic warden or the yard's resident heavy, Dragline, who ends up admiring the new guy's unbreakable will. Luke's bravado, even in the face of repeated stints in the prison's dreaded solitary confinement cell, "the box," make him a rebel hero to his fellow convicts and a thorn in the side of the prison officers.
          </textarea>
        </div>
        <div
          class="action"
        >
          <button
            class="resetBtn"
          >
            RESET
          </button>
          <button
            class="submitBtn"
            type="submit"
          >
            SUBMIT
          </button>
        </div>
      </form>
      <div
        class="close"
      >
        <svg
          fill="none"
          height="30"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M6.2253 4.81108C5.83477 4.42056 5.20161 4.42056 4.81108 4.81108C4.42056 5.20161 4.42056 5.83477 4.81108 6.2253L10.5858 12L4.81114 17.7747C4.42062 18.1652 4.42062 18.7984 4.81114 19.1889C5.20167 19.5794 5.83483 19.5794 6.22535 19.1889L12 13.4142L17.7747 19.1889C18.1652 19.5794 18.7984 19.5794 19.1889 19.1889C19.5794 18.7984 19.5794 18.1652 19.1889 17.7747L13.4142 12L19.189 6.2253C19.5795 5.83477 19.5795 5.20161 19.189 4.81108C18.7985 4.42056 18.1653 4.42056 17.7748 4.81108L12 10.5858L6.2253 4.81108Z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="circle"
        data-testid="circle"
      >
        <div
          class="firstDot"
        />
        <div
          class="secondDot"
        />
        <div
          class="thirdDot"
        />
      </div>
      <img
        alt="Not found"
        class="image"
        src="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
      />
      <div
        class="info"
      >
        <div
          class="title"
        >
          Cool Hand Luke
        </div>
        <div
          class="year"
        >
          1967-06-22
        </div>
      </div>
      <div
        class="genre"
      >
        Crime, Drama
      </div>
    </div>
  </div>
</div>
`;

exports[`Movie component should render Movie component with the movie details 1`] = `
<div>
  <div>
    <div
      class="card"
      data-testid="card"
    >
      <div
        class="circle"
        data-testid="circle"
      >
        <div
          class="firstDot"
        />
        <div
          class="secondDot"
        />
        <div
          class="thirdDot"
        />
      </div>
      <img
        alt="Not found"
        class="image"
        src="https://image.tmdb.org/t/p/w500/vDwphkloD7ToaDpKASAXGgHOclN.jpg"
      />
      <div
        class="info"
      >
        <div
          class="title"
        >
          Cool Hand Luke
        </div>
        <div
          class="year"
        >
          1967-06-22
        </div>
      </div>
      <div
        class="genre"
      >
        Crime, Drama
      </div>
    </div>
  </div>
</div>
`;
